### PR TITLE
Add RemoteDepth to llamahub

### PR DESCRIPTION
**Pull request by simonManydata**
Needed to build a reader where we download all the links available in a webpage (also links inside links).

Difficult to write tests without network requests. 

Not sure if I should have extended the reader `RemoteReader`, modify existing  `RemoteReader`, or just create a new class from BaseClass.

### CODE REVIEW

File: loader_hub/remote_depth/base.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+"""
+from typing import Any, Dict, List, Optional, Union
+from urllib.parse import urljoin, urlparse, urlunparse
+
+import requests
+from bs4 import BeautifulSoup
+from llama_index import download_loader
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+from tqdm.auto import tqdm
```
Comment (emptycrown) [inline loader_hub/remote_depth/base.py:14]: Can you lazily import these packages from within the class? For now, it makes it easier to test and also download many loaders at once. Additionally, please create a `requirements.txt` in this directory and drop in bs4 and tqdm.

File: loader_hub/remote_depth/base.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+
+class RemoteDepthReader(BaseReader):
```
Comment (emptycrown) [inline loader_hub/remote_depth/base.py:17]: I think it's ok to have this as a separate class for now. IMO it's closer to a web scraper than `RemoteReader`.

File: loader_hub/remote_depth/base.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
+"""Remote file reader.
+
+A loader that fetches an arbitrary remote page or file by URL and parses its contents.
```
Comment (emptycrown) [inline loader_hub/remote_depth/base.py:3]: Update this description.

**Review by emptycrown — CHANGES_REQUESTED** [review]
Thanks! Left a few comments but looks good overall.

File: loader_hub/remote_depth/base.py (hunk 2 above)
Comment (simonManydata) [reply loader_hub/remote_depth/base.py:14]: Yes, correct, Should we rename that to `WebScraperReader` when we have more functionalities, and keep that named `RemoteDepthReader` for now?

File: loader_hub/remote_depth/base.py (hunk 1 above)
Comment (simonManydata) [reply loader_hub/remote_depth/base.py]: I have lazily imported the packages `bs4, `tqdm` and `urllib.parse`

**Review by emptycrown — APPROVED** [review]

File: loader_hub/remote_depth/base.py (hunk 2 above)
Comment (emptycrown) [reply loader_hub/remote_depth/base.py:13]: Yeah let's keep it for now. There are other web scrapers in the library already. We can reorganize down the road if needed.

**Comment by simonManydata** [comment]
Still on the 24 from PST!

**Comment by emptycrown** [comment]
Yep you're in the pool :)

**Pull request merged**